### ISSUE TITLE
Bug: [file-extension-in-import] does not work with allowImportingTsExtensions

### ISSUE THREAD

**Issue by Zamiell** [eslint-community/eslint-plugin-n#422]
### Environment

Node version: 22.14.0
npm version: 11.1.0
ESLint version: 9.22.0
eslint-plugin-n version: 17.16.2
Operating System: Windows

See my simple reproduction repository below.

### What rule do you want to report?

file-extension-in-import

### Link to Minimal Reproducible Example

https://github.com/Zamiell/eslint-n-bug

### What did you expect to happen?

no error to occur

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by Zamiell** [comment]
Looks like this feature was implemented by Sebastian Good: https://github.com/eslint-community/eslint-plugin-n/issues/134 / https://github.com/eslint-community/eslint-plugin-n/pull/139

It seems I can't ping him directly here, maybe someone else can?

**Comment by wycats** [comment]
@Zamiell it looks like this was supposed to be fixed in v17. I'm confused... was it not fixed? Do you know of any workaround?

**Comment by wycats** [comment]
Future travelers: I was trying to set `typescriptExtensionMap` at the top-level of `settings` when it needed to be nested under `node`. This configuration works:

```ts
{
  "settings": {
    "node": {
      "typescriptExtensionMap": [
        [
          "",
          ".ts"
        ],
        [
          ".ts",
          ".ts"
        ]
      ]
    }
  }
}
``` 

It's just a workaround, but it works.

**Comment by scagood** [comment]
Sorry, struggling for time at the moment 👀

This was only implemented for "no-missing-*" before (in [lib/util/get-try-extensions.js:65-71](https://github.com/eslint-community/eslint-plugin-n/blob/master/lib/util/get-try-extensions.js#L65-L71))

---

I think `allowImportingTsExtensions` handling needs to be add to [lib/util/get-typescript-extension-map.js:64-72](https://github.com/eslint-community/eslint-plugin-n/blob/master/lib/util/get-typescript-extension-map.js#L64-L72) to get the best effect.

Thanks for the example @wycats :)

---

Im not 100% sure what to do with the `.tsx` extension -- I will have to do some reading.

Currently these are the ones I think are correct:
```
[ ".ts", ".ts" ]
[".cts", ".cts"]
[".mts", ".mts"]
```

I am on the fence about:
```
["", ".ts"]
```
